### ISSUE TITLE
Android dependencies not found during pitest

### ISSUE THREAD

**Issue by lachudra87** [koral--/gradle-pitest-plugin#41]
Hi, 
I use latest pitest version (0.1.7) with android gradle plugin 3.2.0 and receive lot of `ClassDefNotFoundError` errors for android classes. I guess this is result of changes in mockableAndroidJar creation, but any of fixes that were already applied for this issue is working for me.
From my observation it seems that mockable jar is not generated at all, but I am not sure if this is normal behavior for latest gradle plugin. Could you tell me what is current logic for pitest plugin, does it copy android jar itself (I think there was such task at some time) or it base on sth different.

I will try to create minimal project with this problem but it may take some time.

**Comment by lachudra87** [comment]
Some logs
```
> Task :app:pitestDebug
12:51:42 PIT >> INFO : Verbose logging is disabled. If you encounter an problem please enable it before reporting an issue.
12:51:43 PIT >> INFO : Sending 18 test classes to minion
12:51:43 PIT >> INFO : Sent tests to minion
12:51:43 PIT >> INFO : MINION : 12:51:43 PIT >> INFO : Checking environment
12:51:43 PIT >> INFO : MINION : 12:51:43 PIT >> WARNING : Could not load pl.bl.test.pitestbugexample.SearchViewIcon$setOnClickListener$1 (NoClassDefFoundError: android/view/View$OnClickListener)
12:51:43 PIT >> INFO : MINION : 12:51:43 PIT >> INFO : Found  1 tests
12:51:43 PIT >> INFO : MINION : 12:51:43 PIT >> INFO : Dependency analysis reduced number of potential tests by 0
12:51:43 PIT >> INFO : MINION : 12:51:43 PIT >> INFO : 1 tests received
12:51:44 PIT >> INFO : MINION : 12:51:44 PIT >> SEVERE : Description [testClass=pl.bl.test.pitestbugexample.SearchViewIconTest, name=dummy test that should compile(pl.bl.test.pitestbugexample.SearchViewIconTest)]
java.lang.NoClassDefFoundError: android/view/ViewGroup
        at java.lang.Clas
12:51:44 PIT >> INFO : MINION : sLoader.defineClass1(Native Method)
        at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
        at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
        at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
        at java.net.UR
12:51:44 PIT >> INFO : MINION : LClassLoader.access$100(URLClassLoader.java:73)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.net.URLClassLo
12:51:44 PIT >> INFO : MINION : ader.findClass(URLClassLoader.java:361)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        at java.lang.ClassLoader.defi
12:51:44 PIT >> INFO : MINION : neClass1(Native Method)
        at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
        at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
        at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
        at java.net.URLClassLoader
12:51:44 PIT >> INFO : MINION : .access$100(URLClassLoader.java:73)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.net.URLClassLoader.findCla
12:51:44 PIT >> INFO : MINION : ss(URLClassLoader.java:361)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        at pl.bl.test.pitestbugexample.SearchView
12:51:44 PIT >> INFO : MINION : IconTest.dummy test that should compile(SearchViewIconTest.kt:17)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorI
12:51:44 PIT >> INFO : MINION : mpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
        at org.junit.internal.runners.model.ReflectiveCallable.run(Re
12:51:44 PIT >> INFO : MINION : flectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunne
12:51:44 PIT >> INFO : MINION : r.java:325)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
        at
12:51:44 PIT >> INFO : MINION : org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
        at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
        at org.junit.runners.ParentRunner$2.evaluate(Paren
12:51:44 PIT >> INFO : MINION : tRunner.java:268)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
        at org.mockito.internal.runners.DefaultInternalRunner$1.run(DefaultInternalRunner.java:79)
        at org.mockito.internal.runners.DefaultInternalRunner.run(DefaultInternalRunner.
12:51:44 PIT >> INFO : MINION : java:85)
        at org.mockito.internal.runners.StrictRunner.run(StrictRunner.java:39)
        at org.mockito.junit.MockitoJUnitRunner.run(MockitoJUnitRunner.java:163)
        at org.pitest.junit.adapter.CustomRunnerExecutor.run(CustomRunnerExecutor.java:42)
        at org.pites
12:51:44 PIT >> INFO : MINION : t.junit.adapter.AdaptedJUnitTestUnit.execute(AdaptedJUnitTestUnit.java:69)
        at org.pitest.coverage.execute.CoverageDecorator.execute(CoverageDecorator.java:50)
        at org.pitest.testapi.execute.containers.UnContainer.execute(UnContainer.java:31)
        at org.pi
12:51:44 PIT >> INFO : MINION : test.testapi.execute.Pitest.executeTests(Pitest.java:57)
        at org.pitest.testapi.execute.Pitest.run(Pitest.java:48)
        at org.pitest.coverage.execute.CoverageWorker.run(CoverageWorker.java:51)
        at org.pitest.coverage.execute.CoverageMinion.main(CoverageMin
12:51:44 PIT >> INFO : MINION : ion.java:90)
Caused by: java.lang.ClassNotFoundException: android.view.ViewGroup
        at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at sun.misc.Launcher$AppClassLoader.loadClass(Lau
12:51:44 PIT >> INFO : MINION : ncher.java:331)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        ... 54 more
12:51:44 PIT >> SEVERE : Description [testClass=pl.bl.test.pitestbugexample.SearchViewIconTest, name=dummy test that should compile(pl.bl.test.pitestbugexample.SearchViewIconTest)] did not pass without mutation.                            /12:51:44 PIT >> INFO : Calculated coverage in 1 seconds.
Exception in thread "main" org.pitest.help.PitHelpError: All tests did not pass without mutation when calculating line coverage. Mutation testing requires a green suite.
See http://pitest.org for more details.
        at org.pitest.coverage.execute.DefaultCoverageGenerator.verifyBuildSuitableForMutationTesting(DefaultCoverageGenerator.java:109)
        at org.pitest.coverage.execute.DefaultCoverageGenerator.calculateCoverage(DefaultCoverageGenerator.java:94)
        at org.pitest.coverage.execute.DefaultCoverageGenerator.calculateCoverage(DefaultCoverageGenerator.java:49)
        at org.pitest.mutationtest.tooling.MutationCoverage.runReport(MutationCoverage.java:115)
        at org.pitest.mutationtest.tooling.EntryPoint.execute(EntryPoint.java:116)
        at org.pitest.mutationtest.tooling.EntryPoint.execute(EntryPoint.java:49)
        at org.pitest.mutationtest.commandline.MutationCoverageReport.runReport(MutationCoverageReport.java:87)
        at org.pitest.mutationtest.commandline.MutationCoverageReport.main(MutationCoverageReport.java:45)
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':app:pitestDebug'.
```

**Comment by lachudra87** [comment]
[PitestBugExample.zip](https://github.com/koral--/gradle-pitest-plugin/files/2460189/PitestBugExample.zip)
Small example project

**Comment by koral--** [comment]
Thanks, I'll check that.

> Could you tell me what is current logic for pitest plugin, does it copy android jar itself (I think there was such task at some time) or it base on sth different.

JAR is not copied. Plugin just figures out its path and adds it to classpath. Here is the relevant code: https://github.com/koral--/gradle-pitest-plugin/blob/master/src/main/groovy/info/solidsoft/gradle/pitest/PitestPlugin.groovy#L229

**Comment by koral--** [comment]
It seems that since AGP version 3.2.0 the (mockable) Android JAR is not saved into build directory but into Gradle cache using transform.
I'll figure out the way how to get its path.

**Comment by lachudra87** [comment]
We added some temporary workaround for this (task which copy local android jar to build dir) but I would appreciate some proper solution :)

**Comment by koral--** [comment]
I've bumped AGP dependency version to 3.2.1 and added fix for this issue. Commit linked above.
Unfortunately that change breaks compatibility with Robolectric (at least test for that fails).
Moreover, another test which should prevent this issue turns out to be not working as expected, classpath dependencies in functional tests are overridden.
So there is a little bit more work need before releasing version with this fix.

**Comment by koral--** [comment]
Version 0.1.8-SNAPSHOT is on maven snapshot repo.
Robolectric test fails only when using nebula framework. I'll try to change it to TestKit.

**Comment by koral--** [comment]
Sorry for the delay. I've just released version 0.1.8 with this issue fixed and created new one for case with robolectric.
